ext/standard/file.c

```diff
@@ -1669,7 +1669,7 @@ PHP_FUNCTION(copy)
 		Z_PARAM_RESOURCE_EX(zcontext, 1, 0)
 	ZEND_PARSE_PARAMETERS_END();
 
-	if (php_check_open_basedir(source)) {
+	if (php_stream_locate_url_wrapper(source, NULL, 0) == &php_plain_files_wrapper && php_check_open_basedir(source)) {
```

Comment:
If that's the case, shouldn't it be part of php_check_open_basedir itself?

Reply:
Somehow that would be better, but it seems that a lot of related code assumes otherwise, and I'm not sure about the performance impact.

Reply:
Which code assumes otherwise?

Reply:
Bad wording, sorry! I mean that some code already checks whether a plain_files_wrapper is given, and only then calls `php_check_open_basedir()`, e.g. https://github.com/php/php-src/blob/1bc56dbdba68f3729639db584a7d23c710f4b24a/ext/standard/filestat.c#L778

Reply:
@nikic, any suggestions on how to proceed here?